tests/integration/targets/vmware_esxi_maintenance_mode/tasks/eco-vcenter.yml

```diff
@@ -0,0 +1,117 @@
+        password: "{{ vcenter_password }}"
+        datacenter: "{{ vcenter_datacenter }}"
+        cluster: "{{ vcenter_cluster_name }}"
+        validate_certs: false
+        library: "{{ ci_resources_content_library }}"
+        template: "{{ esxi_content_library_template }}"
+        datastore: "{{ shared_storage_02 }}"
+        name: "{{ test_esxi_hostname }}"
+
+    - name: Power On Host
+      community.vmware.vmware_guest_powerstate:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        datacenter: "{{ vcenter_datacenter }}"
+        validate_certs: false
+        state: powered-on
+        name: "{{ test_esxi_hostname }}"
+
+    - name: Wait For ESXi Host To Have IP
+      vmware.vmware.guest_info:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        validate_certs: false
+        name: "{{ test_esxi_hostname }}"
+      register: _esxi_info
+      until: _esxi_info.guests[0].ipv4 is defined and _esxi_info.guests[0].ipv4
+      retries: 10
+      delay: 5
+
+    - name: Join Hosts To Cluster
+      community.vmware.vmware_host:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        validate_certs: false
+        datacenter: "{{ vcenter_datacenter }}"
+        cluster: "{{ vcenter_cluster_name }}"
+        esxi_hostname: "{{ _esxi_info.guests[0].ipv4 }}"
+        esxi_username: root
+        esxi_password: "!#%135qEt"
+        state: present
+
+    - name: Unset Host Maintenance Mode
+      vmware.vmware.esxi_maintenance_mode:
+        validate_certs: false
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        enable_maintenance_mode: False
+        name: "{{ _esxi_info.guests[0].ipv4 }}"
+      register: _set
+    - name: Check Task Output
+      ansible.builtin.assert:
+        that:
+          - _set is not changed
+
+    - name: Set Host Maintenance Mode
+      vmware.vmware.esxi_maintenance_mode:
+        validate_certs: false
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        enable_maintenance_mode: true
+        name: "{{ _esxi_info.guests[0].ipv4 }}"
+      register: _set
+
+    - name: Gather Info about ESXi Host
+      community.vmware.vmware_host_facts:
+        validate_certs: false
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        esxi_hostname: "{{ _esxi_info.guests[0].ipv4 }}"
+      register: _host_info
+
+    - name: Check Task Output
+      ansible.builtin.assert:
+        that:
+          - _set is changed
+          - _host_info.ansible_facts.ansible_in_maintenance_mode
+
```

Comment:
Add task `Unset Host Maintenance Mode For Host That In Maintenance Mode` and check task output please